R/plotPValHist.R

```diff
@@ -25,110 +24,146 @@
-
-                plotlist[[i]] = Hist_Pval
-            }
+            #for (i in 1:samples_num) {
+            plotlist <- lapply(sample_names, function(sample) {
+                pval_subset <- dplyr::filter(P.Val, grepl(sample, levels))
+                cx.data <- pval_subset %>% dplyr::select(pval)
+                var.annot <- pval_subset %>% dplyr::select(-pval)
+                hist_pval <- canvasXpress::canvasXpress(data                 = cx.data,
+                                                        varAnnot             = var.annot,
+                                                        histogramData        = TRUE,
+                                                        histogramBinWidth    = binWidth,
+                                                        graphType            = "Scatter2D",
+                                                        colors               = color,
+                                                        transparency         = transparency,
+                                                        title                = paste(title, "\n", sample),
+                                                        xAxisTitle           = "P-value",
+                                                        yAxisTitle           = "Count",
+                                                        hideHistogram        = FALSE,
+                                                        showHistogramDensity = FALSE,
+                                                        showLegend           = FALSE,
+                                                        events               = events)
+                hist_pval
+            })
         }
     } else {
         if (facet) {
             numcol <- 3
-            numrow <- (NumSamples / numcol) %>% ceiling
+            numrow <- (samples_num / numcol) %>% ceiling
 
-            plotlist <- ggplot2::ggplot(data = P.Val, aes(x = Pval)) +
-                ggplot2::geom_histogram(alpha = alpha,
-                                        fill = fill,
-                                        color = color,
+            plotlist <- ggplot2::ggplot(data = P.Val, aes(x = pval)) +
+                ggplot2::geom_histogram(alpha    = transparency,
+                                        fill     = color,
+                                        color    = color,
                                         binwidth = binWidth) +
                 ggplot2::xlab("P-value") +
                 ggplot2::ylab("Count") +
                 ggtitle(title) +
                 ggplot2::scale_fill_brewer(palette = "Set1") +
-                ggplot2::facet_wrap(~Levels, nrow = numrow, scales = "free")
+                ggplot2::facet_wrap(~levels, nrow = numrow, scales = "free")
         } else {
-            for (i in 1:NumSamples) {
-                s <- SampNames[i]
-                MyPVal <- dplyr::filter(P.Val, grepl(s, Levels))
+            for (i in 1:samples_num) {
```

Comment:
I suggest you use the same code style as in you did in canvasXpress and use lapply instead of the for loop

Reply:
Done